Feature/add env source for credentials (and rename which it's based on, which also has sep feature branch)

bglusman opened the pull request:
@tobowers I think this has you written all over it :-)  Remind me if we were going to add/change anything else besides the heroku env thing.

Commit:
initial file content rename, doing file names separately

Commit:
renamed all files to clusterfsck

Commit:
Merge branch 'feature/rename_project'

Commit:
added from_env_vars method and test

tobowers commented on spec/lib/cluster-fsck/credential_grabber_spec.rb:
```diff
@@ -58,17 +65,34 @@ module ClusterFuck
 
+    describe "when there is both ENV vars and a ~/.fog file" do
+      before :each do
+        ENV.stub(:[]).with("AWS_ACCESS_KEY_ID").and_return('env_access_key')
```
this should use the env_credentials['AWS_ACCESS_KEY_ID'] from the let above

Commit:
rename files for project rename

Commit:
move to non-amicus specific config, ask for bucket

tobowers commented on lib/cluster-fsck.rb:
```diff
@@ -0,0 +1,54 @@
+  end.compact.first || {}
+
+  CONFIG_BUCKET = CLUSTER_FSCK_CONFIG['BUCKET'] || SETUP_BUCKET.call
+  CLUSTER_FSCK_ENV = CLUSTER_FSCK_CONFIG['ENV'] || 'development'
```
should be able to set this with an environment variable too (as one can now).  Should probably be able to skip all the files if you used an env variable... as in... the bucket should be settable there too.

Commit:
clarify env usage

Commit:
Fix config and tests, allow env to override settings

Commit:
get rid of some constants

Commit:
prompt for S3 keys and/or for bucket

Commit:
Check for bucket existence and offer to create

Commit:
moving help text from blogpost to readme

Commit:
fix regression bug on warn-create-bucket

Commit:
Add bin/alias with dash for CLI

Commit:
This was illconceived, shouldn't do with load anyway, removing for now.

This reverts commit f0adac012672121f60f5376589f91f0b27b4562c.

bglusman commented:
Just a ping that this is still waiting on review/merge, though getting everything over to use this version may be annoying at this point, dunno...

bglusman commented:
Er, forget if that pings you without mentioning @tobowers sorry if you get twice :-)

Commit:
refactor/extract setup to new module

was going to add test coverage, but stubbing:logic
ratio seems way off, stub lots to test like, an if
statement, so leaving untested for now…
Tested manually though.
